OCP-4.X/roles/winc-install/tasks/main.yml

```
@@ -0,0 +1,104 @@
+# Applies Windows Worker nodes configuration to a cluster post installation
+---
+#  These registers be set from post-install tasks, but if you skip those tasks, you can enable them here to run this role one-off.
+# - name: Get cluster name
+#   shell: |
+#     {%raw%}oc get machineset -n openshift-machine-api -o=go-template='{{(index (index .items 0).metadata.labels {%endraw%} "{{ machineset_metadata_label_prefix }}/cluster-api-cluster" {%raw%})}}'{%endraw%}
+#   register: cluster_name
+#   environment:
+#     KUBECONFIG: "{{ kubeconfig_path }}"
+
+# - name: (AWS) set cluster region
+#   shell: |
+#     {%raw%}oc get machineset -n openshift-machine-api -o=go-template='{{(index .items 0).spec.template.spec.providerSpec.value.placement.region}}'{%endraw%}
+#   register: aws_region
+#   environment:
+#     KUBECONFIG: "{{ kubeconfig_path }}"
+
+- name: winC AWS - Get Windows AMI ID
+  shell: |
+    aws ec2 describe-images --filters Name=name,Values={{ primary_windows_image }}* --region {{ aws_region.stdout }} --query {%raw%}'sort_by(Images, &CreationDate)[-1].[ImageId]'{%endraw%} --output text
+  register: windows_ami_id
+  when: platform == "aws"
+
+- name: winC - Create WMCO catalog source
+  kubernetes.core.k8s:
+    kubeconfig: "{{ kubeconfig_path }}"
+    name: openshift-windows-machine-config-operator
+    template: catalogsource.yaml.j2
+  retries: 3
+
+- name: winC - Wait for catalog source
+  shell: oc get catalogsource wmco -o=jsonpath={.status.connectionState.lastObservedState} -n openshift-marketplace
+  environment:
+    KUBECONFIG: "{{ kubeconfig_path }}"
+  register: result
+  until: result.stdout == "READY"
+  retries: 3
+  delay: 10
+
+- name: winC - Create WMCO namespace
+  kubernetes.core.k8s:
+    kubeconfig: "{{ kubeconfig_path }}"
+    state: present
+    api_version: v1
+    kind: Namespace
+    name: openshift-windows-machine-config-operator
+
+- name: winC - Add monitoring label to wmco namespace
+  kubernetes.core.k8s:
+    kubeconfig: "{{ kubeconfig_path }}"
+    state: patched
+    kind: Namespace
+    name: openshift-windows-machine-config-operator
+    definition:
+      metadata:
+        labels:
+          openshift.io/cluster-monitoring: "true"
```

You can merge these tasks into one with something like

```suggestion
- name: Create WMCO namespace
  kubernetes.core.k8s:
    kubeconfig: "{{ kubeconfig_path }}"
    state: present
    definition:
      kind: Namespace
      apiVersion: v1
      metadata:
        name: openshift-windows-machine-config-operator
        labels:
          openshift.io/cluster-monitoring: "true"
```